src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -170,14 +179,15 @@ Copyright (c) .NET Foundation. All rights reserved.
   
   <Target Name="ResolvePackageDependenciesDesignTime"
           Returns="@(_DependenciesDesignTime)"
-          DependsOnTargets="ResolveAssemblyReferencesDesignTime;RunResolvePackageDependencies">
+          DependsOnTargets="ResolveAssemblyReferencesDesignTime;RunResolvePackageDependencies;$(EmitsDependencyDiagnosticMessages)">
```

Comment:
Why is RunResolvePackageDependencies hard-coded here and also in EmitsDependencyDiagnosticMessages?